initfiles/componentfiles/configschema/xsd/esp.xsd

```diff
@@ -145,91 +214,176 @@
+                                          hpcc:tooltip="whether to accept self-signed certificates"/>
+                            <xs:attribute name="CA_Certificates_Path" type="xs:string" use="optional"
+                                          hpcc:presetValue="ca.pem" hpcc:displayName="Certificates Path"
+                                          hpcc:requiredIf="../EspBinding[@protocol='https']"
                                           hpcc:tooltip="path to the file that contains CA certificates"/>
-                            <xs:attribute name="certificateFileName" type="xs:string" use="optional" hpcc:presetValue="certificate.cer" hpcc:displayName="Certificate Filename" hpcc:tooltip="Name of destination file in which the certificate will be written"/>
-                            <xs:attribute name="cipherList" type="xs:string" use="optional" hpcc:presetValue="ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
-                                          hpcc:displayName="Cipher List" hpcc:tooltip="Sets the ordered list of available ciphers for use by openssl.  See openssl documentation on ciphers for information on use and formatting."/>
-                            <xs:attribute name="passphrase" type="xs:string" use="optional" hpcc:modifiers="mask,verify,ignoreNoChange,encrypt" hpcc:displayName="Passphrase" hpcc:tooltip="The passphrase used to generate the private key" />
-                            <xs:attribute name="privateKeyFileName" type="xs:string" use="optional" hpcc:presetValue="privatekey.cer" hpcc:displayName="Private Key Filename" hpcc:requiredIf="../EspBinding[@protocol='https']"
-                                          hpcc:tooltip="Name of destination file in which the private key will be written" />
+                            <xs:attribute name="certificateFileName" type="xs:string" use="optional"
+                                          hpcc:presetValue="certificate.cer" hpcc:displayName="Certificate Filename"
+                                          hpcc:tooltip="Name of destination file in which the certificate will be written"/>
+                            <xs:attribute name="cipherList" type="xs:string" use="optional"
+                                          hpcc:presetValue="ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
+                                          hpcc:displayName="Cipher List"
+                                          hpcc:tooltip="Sets the ordered list of available ciphers for use by openssl.  See openssl documentation on ciphers for information on use and formatting."/>
+                            <xs:attribute name="passphrase" type="xs:string" use="optional"
+                                          hpcc:modifiers="mask,verify,ignoreNoChange,encrypt"
```

Comment:
Can you explain each of: mask,verify,ignoreNoChange,encrypt?

Reply:
Those are modifiers for the UI. They are covered on the Wiki page, at least how modifiers are handled. Basically it is a way for the UI to encode information about how it may display the setting. The core library doesn't use them, only passes them through via the ESP interface.

Reply:
@wangkx Additionally, it is likely these values will change when Miguel starts to work on the UI and begins to handle fields like this.